src/flow/push_relabel.jl

```diff
@@ -209,27 +209,28 @@ vertex if the excess remains non-zero.
 Requires arguments:
 
 - residual_graph::DiGraph                 # the input graph
-- v::Int                                  # vertex to be discharged
+- v::Integer                              # vertex to be discharged
 - capacity_matrix::AbstractArray{T,2}
 - flow_matrix::AbstractArray{T,2}
 - excess::AbstractArray{T,1}
 - height::AbstractArray{Int,1}
 - active::AbstractArray{Bool,1}
 - count::AbstractArray{Int,1}
-- Q::AbstractArray{Int,1}
+- Q::AbstractArray{Integer,1}
 """
-function discharge!{T<:Number}(
-    residual_graph::DiGraph,                # the input graph
-    v::Int,                                 # vertex to be discharged
-    capacity_matrix::AbstractArray{T,2},
-    flow_matrix::AbstractArray{T,2},
-    excess::AbstractArray{T,1},
-    height::AbstractArray{Int,1},
-    active::AbstractArray{Bool,1},
-    count::AbstractArray{Int,1},
-    Q::AbstractArray{Int,1}                 # FIFO queue
+function discharge! end
+@traitfn function discharge!{G<:AbstractGraph; IsDirected{G}}(
+    residual_graph::G,                # the input graph
+    v::Integer,                                 # vertex to be discharged
+    capacity_matrix::AbstractMatrix,
+    flow_matrix::AbstractMatrix,
+    excess::AbstractVector,
```

Comment:
I don't know - my goal was to get things "working". Let's table this for now unless there's some obvious problem?